### ISSUE TITLE
[Dialog] Content should not have the `linear-gradient` applied when no scrollbar present

### ISSUE THREAD

**Issue by chris-griffin** [narmi/design_system#1126]
### Component/Module
Dialog

### Screenshots (if applicable)

The fading in the "profile?" text below is not accessible

![image](https://github.com/narmi/design_system/assets/1795659/34c5d80b-7a60-4283-9220-e2d9a5b7d8ba)

It goes away when removing the background-image rule

![image](https://github.com/narmi/design_system/assets/1795659/fde57880-93e6-43ba-8cf5-f36e567adbfd)

### Reproduction Steps

Render a Dialog with limited text so a scrollbar is not present

### Expected Result

The text does not fade when there is no scrollbar for the Dialog content

### Actual Result

The text fades always

**Comment by akdetrick** [comment]
The overflow styles should not be applied to a Dialog this short. Checking in `banking` to try to reproduce the issue
